docs/features/sinks/azure-application-insights.md

```diff
@@ -57,4 +57,66 @@ if(string.IsNullOrEmpty(key) == false)
 ILogger logger = loggerConfig.CreateLogger();
 ```
 
+### Q: Where can I initialize the logger in an ASP.NET Core application or other hosted service?
+
+The Application Insights instrumentation key is typically available via the configuration or via an `ISecreteProvider`.  It's best if the Logger is created and assigned to the Serilog `Log.Logger` as soon as possible.  The easiest place to do that in an ASP.NET application  is in the `ConfigureServices` method of the `Program` class.
```

Comment:
```suggestion
The Azure Application Insights instrumentation key is typically available via the configuration or `ISecreteProvider`.  It's best if the logger is created and assigned to the Serilog `Log.Logger` as soon as possible.

The easiest place to do that in an ASP.NET application is in the `ConfigureServices` method of the `Program` class.
```

However, we actually do it as part of UseSerilog: https://github.com/arcus-azure/arcus.templates/blob/master/src/Arcus.Templates.ServiceBus.Queue/Program.cs#L74 but use a start-up logger to catch all logs before that: https://github.com/arcus-azure/arcus.templates/blob/master/src/Arcus.Templates.ServiceBus.Queue/Program.cs#L31-L34

Reply:
With the current proposal you can't use anything from the secret store

Reply:
Oh, so you have an overload UseSeriLog method which allows you to configure the logger ?  That's actually even better.

Reply:
> With the current proposal you can't use anything from the secret store

Can you please elaborate ?

Reply:
No, more because it's all part of `ConfigureAppConfiguration`